src/Api.php

```diff
@@ -185,15 +208,23 @@ protected function call($callable, $vars = [])
      *
      * Extend this method to implement your own field restrictions.
      *
-     * @param Model $m
+     * @param Model $m              Model
+     * @param array $allowed_fields Allowed fields
      *
      * @throws \atk4\data\Exception
+     * @throws \atk4\core\Exception
      *
      * @return array
      */
-    protected function exportModel(Model $m)
+    protected function exportModel(Model $m, array $allowed_fields = [])
     {
-        return $m->export($this->getAllowedFields($m, 'read'));
+        $data = [];
+        foreach ($allowed_fields as $fieldName) {
+            $field = $m->getField($fieldName);
+            $data[$field->actual ?? $fieldName] = $field->toString();
```

Comment:
I dont think using actual is sensible. Think of this example Model:

```
$this->addField('name');
$join = $this->join('some_table');
$join->addField('other_name', ['actual' => 'name']);
```

That would lead to replacing the name of the model's name field with the name field from the joined table, which was sensibly declared in the model as `other_name`

Reply:
In the demo there was a Model with few fields with actual defined and even if i don't use it i try to fix the right field name to return back in api response

Reply:
Thinking of it this way:
A Model can have each field name only once. So no trouble using this as array key.
When using `actual` property, we can have duplicate array keys, which leads to false/incomplete data passed via API.
I think `actual` is meant for Persistences only.